les/client.go

```diff
@@ -140,6 +132,15 @@ func New(stack *node.Node, config *eth.Config) (*LightEthereum, error) {
 	leth.chainReader = leth.blockchain
 	leth.txPool = light.NewTxPool(leth.chainConfig, leth.blockchain, leth.relay)
 
+	discovery, err := leth.setupDiscovery(&stack.Config().P2P)
+	if err != nil {
+		return nil, err
+	}
+	leth.serverPool = newServerPool(lespayDb, []byte("serverpool:"), leth.valueTracker, discovery, time.Second, nil, &mclock.System{}, config.UltraLightServers)
+	peers.subscribe(leth.serverPool)
+	leth.dialCandidates = leth.serverPool.dialIterator
+	leth.retriever.softRequestTimeout = leth.serverPool.getTimeout
```

Comment:
If we don't change the New function of retriever, we can get rid of this?

Reply:
Now `setupDiscovery` runs in `Start()` so this change is no longer necessary, I reverted it.